tests/unit_tests/serialization.cpp

```diff
@@ -616,6 +616,25 @@ TEST(Serialization, serializes_ringct_types)
   ASSERT_EQ(bp0, bp1);
 }
 
+TEST(Serialization, key_encryption_transition)
+{
+  const cryptonote::network_type nettype = cryptonote::TESTNET;
+  tools::wallet2 w(nettype);
+  const boost::filesystem::path wallet_file = unit_test::data_dir / "wallet_9svHk1";
+  string password = "test";
+  bool r = false;
+  try
+  {
+    w.load(wallet_file.string(), password); // old key method
+    w.rewrite(wallet_file.string(), password);
+    w.load(wallet_file.string(), password); // new key method
```

Comment:
It might be good to assert that you get the same keys post-rewrite and load as you did the first time around.

Reply:
I suppose this could be specifically tested. However, the load process does check that the wallet address corresponds correctly to the decrypted keys, so the odds that this would pass with invalid keys is negligible.

A better test would be to ensure that the new method is actually being used, rather than falling back to the legacy method in the event that the keys weren't properly rewritten; however, that would require passing information back from the loader, and it's not clear whether that would cause other problems elsewhere.

Reply:
Checking that the new method is used would be reasonable, although you could also just confirm that the serialized representation has changed, which seems good enough.

As for checking the key post-load, I wouldn't want to rely on the implementation detail that the wallet asserts the sanity of the data, unless there's a separate test which explicitly confirms that behavior, which I don't see in this suite. The prior deserialization test, `portability_wallet`, goes to great lengths to validate all the loaded key material, so it seems appropriate to at least confirm that loading an upgraded wallet produces, e.g., the proper public address, as, at the moment, all the prior test suite sanity checks don't happen for the new (and now default) wallet format. In other words, as it stands, for the common case of "create new format wallet, load new format wallet", there's nothing in this test suite confirming data sanity, as was done for the earlier format.

Reply:
I'm already making changes to explicitly check the decryption method used, so I can further add a check that the keys reconstruct the proper address.